### DISCUSSION TITLE
MAPL 2.22.0

### DISCUSSION THREAD

**Discussion by mathomp4** [GEOS-ESM/MAPL#1585] [Announcements]
This is a zero-diff minor release of MAPL. The main changes in these release are: 

- Fix in pfio when missing values are NaN
- Fix in History when trying to write to a file that already exists
- CMake updates for better building with Spack
- Added monotonic regridding option
- Various CI improvements

---

### Fixed

- By pass the check of the missing value of Nan
- Update CI to work with latest GEOSadas `develop` (Uses a special branch of GEOSadas)
- Fix bundleio tests
- HistoryGridComp now checks if a file exists already before writing and errors out if so
- Minor updates for FORD documentation testing
  - Add `program` statements to some test programs
  - Remove `pfio/pfio_io_demo.F90` as dead code
  - Fix redefinition of `_RETURN` in `pflogger_stub.F90`
  - Removed unused `Test_SimpleClient.pf`
- Update CMake to require NetCDF C components and add `NetCDF::NetCDF_C` to pfio CMake

### Added

- Updated MAPL_CapGridComp to mark fields going to ExtData to not be checkpointed by components
- Add debug loggers for start/stop during stages in MAPL_Generic
- Handling for double precision input when retrieving single precision attributes
- Enable GCM run test in CircleCI (1-hour, no ExtData)
- Added monotonic regridding option
- Make available to History and ExtData2G all supported regridding methods
- Add test cases for ExtData
- Add YAML validator GitHub Action
  - This action makes sure all YAML files are valid (to a relaxed standard)

### Changed

- Modified error messages in FileMetadataUtilities to be unique and print filename
- Updated the ESMA_env version to v3.14.0
- Updated the ESMA_cmake version to v3.17.0
- Updated GitHub Actions MAPL build tests
- Added assert for missing file with ExtData2G
- Re-enable bundleio tests in CI
- Updated CircleCI to use latest Baselibs
- Updates for Spack support
   - Add `find_package(MPI)` for non-Baselibs builds
   - Add explicit interface dependence of `MPI` for `ESMF` target
   - Add `esmf` alias library for `ESMF` for compatibility

---

## What's Changed
* Fix ADAS CI by @mathomp4 in https://github.com/GEOS-ESM/MAPL/pull/1509
* Add loggers for generic stages by @mathomp4 in https://github.com/GEOS-ESM/MAPL/pull/1510
* Expand allowed _FillValue type and error handling in FileMetadataUtilities.F90 by @lizziel in https://github.com/GEOS-ESM/MAPL/pull/1506
* Enable GCM run in CI by @mathomp4 in https://github.com/GEOS-ESM/MAPL/pull/1466
* Automatic restart skip by @bena-nasa in https://github.com/GEOS-ESM/MAPL/pull/1516
* GitFlow: Handmerge main into develop 2022-May-24 by @mathomp4 in https://github.com/GEOS-ESM/MAPL/pull/1525
* Monotonic regridding by @bena-nasa in https://github.com/GEOS-ESM/MAPL/pull/1528
* GitFlow: Handmerge main into develop (2022-May-31) by @mathomp4 in https://github.com/GEOS-ESM/MAPL/pull/1534
* Update to ESMA_cmake v3.16.0 by @mathomp4 in https://github.com/GEOS-ESM/MAPL/pull/1540
* Fixes #1544. Adds _ASSERT for missing ExtData file by @mathomp4 in https://github.com/GEOS-ESM/MAPL/pull/1545
* Handmerge main into develop - 2022-Jun-07 by @mathomp4 in https://github.com/GEOS-ESM/MAPL/pull/1553
* Fix bundleio tests by @bena-nasa in https://github.com/GEOS-ESM/MAPL/pull/1559
* Enable bundleio tests in CI by @mathomp4 in https://github.com/GEOS-ESM/MAPL/pull/1561
* fixes #1565 by @bena-nasa in https://github.com/GEOS-ESM/MAPL/pull/1566
* Add ExtData Testing Framework by @bena-nasa in https://github.com/GEOS-ESM/MAPL/pull/1556
* Updates for FORD Testing by @mathomp4 in https://github.com/GEOS-ESM/MAPL/pull/1570
* Update components.yaml to latest env and cmake by @mathomp4 in https://github.com/GEOS-ESM/MAPL/pull/1572
* More Updates for Spack/non-Baselibs Compatibility by @mathomp4 in https://github.com/GEOS-ESM/MAPL/pull/1550
* Add NetCDF C requirement to CMake by @mathomp4 in https://github.com/GEOS-ESM/MAPL/pull/1578
* by pass missing value of NaN by @weiyuan-jiang in https://github.com/GEOS-ESM/MAPL/pull/1576
* Add YAML Linter by @mathomp4 in https://github.com/GEOS-ESM/MAPL/pull/1580
* Update CHANGELOG and CMakeLists for 2.22.0 Release by @mathomp4 in https://github.com/GEOS-ESM/MAPL/pull/1583
* Merge develop into main for 2.22.0 release by @mathomp4 in https://github.com/GEOS-ESM/MAPL/pull/1584

**Full Changelog**: https://github.com/GEOS-ESM/MAPL/compare/v2.21.3...v2.22.0

<hr /><em>This discussion was created from the release <a href='https://github.com/GEOS-ESM/MAPL/releases/tag/v2.22.0'>MAPL 2.22.0</a>.</em>

**Comment by mathomp4** [comment]
@jkbk2004 This release has the NaN fix from @weiyuan-jiang 

@climbfuji This release should have all the latest Spack updates

**Reply by jkbk2004** [reply]
Awesome! Really appreciate for the help.

**Reply by mathomp4** [reply]
@jkbk2004 One note. If you ever update the ESMA_cmake you use on Orion, you'll have to load the `cmakemodules` module. The reason is that we no longer supply `FindESMF.cmake` in ESMA_cmake. But your `cmakemodules` module does.

And one more note. If you ever upgrade to ESMF v8.3.0 (stable), you'll have to update the `FindESMF.cmake` file in `cmakemodules` as you need a slightly newer one from ESMF for things to work.
